enterprise/com/src/main/java/org/neo4j/com/storecopy/StoreCopyServer.java

```diff
@@ -148,15 +152,20 @@ public RequestContext flushStoresAndStreamStoreFiles( String triggerName, StoreW
     {
         try
         {
-            monitor.startTryCheckPoint();
-            long lastAppliedTransaction = checkPointer.tryCheckPoint( new SimpleTriggerInfo( triggerName ) );
-            monitor.finishTryCheckPoint();
-            ByteBuffer temporaryBuffer = ByteBuffer.allocateDirect( (int) ByteUnit.mebiBytes( 1 ) );
+            ThrowingAction<IOException> checkPointAction = () -> {
+                monitor.startTryCheckPoint();
+                checkPointer.tryCheckPoint( new SimpleTriggerInfo( triggerName ) );
+                monitor.finishTryCheckPoint();
+            };
 
             // Copy the store files
-            monitor.startStreamingStoreFiles();
-            try ( ResourceIterator<StoreFileMetadata> files = dataSource.listStoreFiles( includeLogs ) )
+            long lastAppliedTransaction;
+            try ( Resource lock = mutex.storeCopy( checkPointAction );
+                    ResourceIterator<StoreFileMetadata> files = dataSource.listStoreFiles( includeLogs ) )
             {
+                lastAppliedTransaction = checkPointer.lastCheckPointedTransactionId();
+                monitor.startStreamingStoreFiles();
+                ByteBuffer temporaryBuffer = ByteBuffer.allocateDirect( (int) ByteUnit.mebiBytes( 1 ) );
```

Comment:
correct, only saw that only because that line was moved a bit; completely unrelated